src/features/SingleProductPage.js

```diff
@@ -1,14 +1,78 @@
 import React from "react";
 import { useParams } from "react-router-dom";
 import { useGetProductQuery } from "./api/apiSlice";
+import { useState, useEffect } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { productQuantitySetTo } from "./cart/cartSlice";
 
 export default function SingleProductPage() {
   const { productId } = useParams();
+  const dispatch = useDispatch();
   const { data: product, isLoading } = useGetProductQuery(productId);
+  const [amountOfItems, setAmountOfItems] = useState(1);
+  const cart = useSelector((state) => state.cart);
+
+  useEffect(() => {
+    if (product) {
```

Comment:
it's wrong but not necessary for the amountOfItems to wait for the product object itself exist, productId alone is enough.
so the condition on product and the dependency on product of the useEffect is superfluous

Reply:
makes sense!